app/models/node.rb

```diff
@@ -715,6 +718,23 @@ def self.new_note(params)
     [saved, node, revision]
   end
 
+  def self.new_preview_note(params)
+
+    author = User.find(params[:uid])
+    node = Node.new(uid:     author.uid,
+                    title:   params[:title],
+                    comment: 2,
+                    type:    'note')
+    node.status = 4 if author.first_time_poster
+    node.draft if params[:draft] == "true"
+    
+    if params[:main_image] && (params[:main_image] != '')
+      img = Image.find params[:main_image]
+    end
+    
```

Comment:
We could also add the revision definition, because when it is defined in the new_note, it uses 
```
revision = node.new_revision(uid:   author.uid,
                                     title: params[:title],
                                     body:  params[:body])
```

And looking at `new_revision` it does not save the revision in the database;
```
  def new_revision(params)
    title = params[:title] || self.title
    Revision.new(nid: id,
                 uid: params[:uid],
                 title: title,
                 body: params[:body])
  end
```

That way, we still have the params title, and the revision defined.. 
and have 
```suggestion
    [node, img, revision] 
```

at the end

Reply:
Sure @RuthNjeri , sounds good! Feel free to change the code. Using this PR I just wanted to give an example of how we can mock model functions 😄